### PR TITLE
Add tests with GTest for infoPanel

### PR DESCRIPTION
**Pull request by introkun**

### CODE REVIEW

**Review by Aemiii91 — CHANGES_REQUESTED** [review]
Please move tests out of `src` to `/test` and create a separate `/build_test` folder instead of polluting `/build`

File: test/Makefile [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+OFILES := $(OFILES) ../src/infoPanel/imagesCache.o
+include ../src/common/config.mk
+
+TARGET = test_infoPanel
+LDFLAGS := $(LDFLAGS) -lSDL_image -lSDL -lgtest -lgtest_main -lpthread
+
+include ../src/common/commands.mk
+include ../src/common/recipes.mk
+
+../src/infoPanel/imagesCache.o: ../src/infoPanel/imagesCache.c
```
Comment (introkun) [inline test/Makefile:11]: @Aemiii91 can we find a way to handle this by recipes.mk instead automatically (%.o target is not working for me)

Comment (Aemiii91) [reply test/Makefile:11]: Take a look at how I'm doing it here: https://github.com/OnionUI/Onion/blob/new-build-structure/src/gameSwitcher/Makefile

File: test/Makefile [hunk 2]
```diff
@@ -0,0 +1,10 @@
+TEST = 1
+INCLUDE_UTILS = 0
+CFILES := ../src/infoPanel/imagesCache.c
+include ../src/common/config.mk
+
+TARGET = test_infoPanel
```
Comment (Aemiii91) [inline test/Makefile:6]: How are we supposed to use this for multiple tests?

Comment (introkun) [reply test/Makefile:6]: Lets rename the target to something abstract

Comment (Aemiii91) [reply test/Makefile:6]: Oh okay, so this Makefile is for building all tests?

Comment (introkun) [reply test/Makefile:6]: Yes, correct

**Review by Aemiii91 — APPROVED** [review]

**Pull request merged**